build.sbt

```diff
@@ -14,30 +14,45 @@
  * limitations under the License.
  */
 
+/***************************
+* Spark Packages settings *
+***************************/
+ 
+sparkVersion := "2.4.2"
+
 name := "hyperspace-core"
 
+spName := "hyperspace-core"
+
+spAppendScalaVersion := true
+
+spIncludeMaven := true
+
+spIgnoreProvided := true
+
+packageBin in Compile := spPackage.value
+
 lazy val scala212 = "2.12.8"
 lazy val scala211 = "2.11.12"
 lazy val supportedScalaVersions = List(scala212, scala211)
-
-lazy val sparkVersion = "2.4.2"
+lazy val spkVer = "2.4.2"
 
 scalaVersion := scala212
 
 crossScalaVersions := supportedScalaVersions
 
 libraryDependencies ++= Seq(
-  "org.apache.spark" %% "spark-sql" % sparkVersion % "provided" withSources(),
-  "org.apache.spark" %% "spark-core" % sparkVersion % "provided" withSources(),
-  "org.apache.spark" %% "spark-catalyst" % sparkVersion % "provided" withSources(),
+  "org.apache.spark" %% "spark-sql" % spkVer % "provided" withSources(),
```

Comment:
you can just use `sparkVersion.value` instead of this

Reply:
Updated thanks.